src/backend/toolmanager/toolmanager.ts

```diff
@@ -0,0 +1,346 @@
+import { LogPrefix, logWarning } from './../logger/logger'
+import * as axios from 'axios'
+import {
+  ProgressInfo,
+  Repositorys,
+  State,
+  VersionInfo
+} from '../../common/types/toolmanager'
+import * as crypto from 'crypto'
+import {
+  existsSync,
+  mkdirSync,
+  readFileSync,
+  rmSync,
+  statSync
+} from 'graceful-fs'
+
+import {
+  WINEGE_URL,
+  SODA_BOTTLES_URL,
+  DXVK_URL,
+  DXVK_ASYNC_URL,
+  VKD3D_URL,
+  DXVK_NVAPI_URL,
+  PROTONGE_URL
+} from './constants'
+
+import {
+  downloadFile,
+  fetchReleases,
+  getFolderSize,
+  unlinkFile,
+  unzipFile
+} from './utils'
+
+interface getVersionsProps {
+  repositorys?: Repositorys[]
+  count?: number
+}
+
+/**
+ * Fetch all available releases for given {@link Repositorys}.
+ * If no repository is given, all {@link Repositorys} are checked.
+ * @param repositorys array of {@link Repositorys}.
+ * @defaultValue all {@link Repositorys}
+ * @param count max versions to fetch for each {@link Repository}
+ * @defaultValue 100
+ * @returns * resolves with an array of {@link VersionInfo}
+ *          * rejects with an {@link Error}
+ */
+async function getAvailableVersions({
+  repositorys = [
+    Repositorys.WINEGE,
+    Repositorys.PROTONGE,
+    Repositorys.SODA_BOTTLES,
+    Repositorys.DXVK,
+    Repositorys.DXVK_ASYNC,
+    Repositorys.DXVK_NVAPI,
+    Repositorys.VKD3D
+  ],
+  count = 100
+}: getVersionsProps): Promise<VersionInfo[]> {
+  const releases: Array<VersionInfo> = []
+  for await (const repo of repositorys) {
+    switch (repo) {
+      case Repositorys.WINEGE: {
+        await fetchReleases({
+          url: WINEGE_URL,
+          type: 'Wine-GE',
+          count: count
+        })
+          .then((fetchedReleases: VersionInfo[]) => {
+            releases.push(...fetchedReleases)
+          })
+          .catch((error: Error) => {
+            throw error
```

Comment:
do we want to throw an error here? like... what if a user only cares about updating the wine versions? if the dxvk versions fails I understand that would prevent any other tool list to be updated

maybe we should just log that the request to fetch the new versions failed and continue

Reply:
Good point. Actually this is the old code from heroic-wine-downloader and i don't want to change things in this PR. We should do it in small PR's after this is merged.